dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,203 @@
+isolation, breach in one of the components (like USB driver, network stack)
+wouldn't compromise entire OS contrary to monolithic kernels.
+
+#### Running in DLME
+
+seL4 is a secure L4 family microkernel written in C. It has strong security
+guarantees assured by [Formal proofs](https://sel4.systems/Info/FAQ/proof.pml).
+However, these proofs are still incomplete for x86, see
+[Supported Platforms](https://docs.sel4.systems/Hardware) for an up-to-date
+verification status. seL4, due to its microkernel nature, provides higher
+isolation. A breach in one of the components (like USB driver, network stack)
+wouldn't compromise the entire OS, contrary to monolithic kernels.
+
+#### Fobnail integration
+
+seL4 provides virtually no drivers, except a few drivers listed here.
+According to the page linked above, `libusbdrivers` is inactive and lacks XHCI
+support. seL4 has basic support for an old version of musl C library (v1.1.16).
+
+Using seL4 would require a significant amount of work:
+- Extending USB drivers
+- Implementing USB EEM driver
+- Implementing TPM driver
+- Porting [fobnail-attester](https://github.com/fobnail/fobnail-attester)
+- seL4 runs all processes in unprivileged mode, so the kernel itself would have
+  to be modified to allow booting of another kernel
+
+[Genode](https://github.com/genodelabs/genode) can run on seL4 and provides
+drivers, at least for some devices and a network stack. Its support for seL4
+used to be incomplete, and many components were broken. However, that might have
+changed, and Genode may be an easier way to get software running on seL4.
+
+### Linux
+
+In the case of Linux, a minimal distribution will be prepared that meets the
+requirements of the project. [Yocto Project](https://www.yoctoproject.org/) will
+be used for this, because it gives a lot of freedom in manipulating the elements
+that make up the target system.
+
+#### Running in DLME
+
+Ensuring that the operating system works in DLME can be achieved by using the
+[TrenchBoot](https://trenchboot.org/) project. For this purpose, a Yocto
+[meta-fobnail](https://github.com/fobnail/meta-fobnail) layer has been created
+that integrates the necessary elements. The effects of the tests are presented
+in a [separate document](./running-os-in-dlme.md).
+
+#### Fobnail integration
+
+At this point, the [fobnail-attester](https://github.com/fobnail/fobnail-attester)
+application is strongly dependent on Linux. Mainly due to the fact that its
+development took place on this operating system. Therefore, it is important that
+the created minimal OS also has an integrated attester application. The main
+dependencies of its operation are drivers for USB, USB EEM and TPM, but running
+them under Linux will not cause major problems.
+
+## PoC test
+
+Running DLME requires GRUB from TrenchBoot as mainline doesn't have DLME
+support. During PoC we use GRUB from
+[here](https://github.com/3mdeb/grub/tree/tb_xen).
```

Comment:
We know GRUB2 maintainer so why we don't ask, when things will change?

Reply:
According to Daniel we are stuck with discussion how TrenchBoot should be implemented in Linux kernel. Unless this is solved no investment in TrenchBoot GRUB2 implementation would be made.

Reply:
@pietrushnic thanks. @arturkow2000 please, lets mention that in this report.

Reply:
Added